### PR TITLE
Fix environment pull request hasSCM parameter

### PR DESCRIPTION
**Pull request by sanketjpatel**
The bitbucketserver git urls are of the form
http://test-user@example.bitbucketserver.com/scm/test-org/node-http.git

while others are of the form https://github.com/jenkins-x/jx.git

I think I took a wrong approach to solve this problem by introducing the `hasSCM` parameter to the `parsePath` function - https://github.com/jenkins-x/jx/blob/master/pkg/gits/git_url.go#L156

```
func parsePath(path string, hasSCM bool, info *GitRepositoryInfo) (*GitRepositoryInfo, error) {
	arr := strings.Split(strings.TrimPrefix(path, "/"), "/")
	if len(arr) >= 2 {
		if hasSCM {
			info.Organisation = arr[1]
			info.Project = arr[1]
			info.Name = strings.TrimSuffix(arr[2], ".git")
		} else {
			info.Organisation = arr[0]
			info.Name = strings.TrimSuffix(arr[1], ".git")
		}

		return info, nil
	} else {
		return info, fmt.Errorf("Invalid path %s could not determine organisation and repository name", path)
	}
}
```
Could have avoided sending the hasSCM parameter and updated the method as follows:

```
func parsePath(path string, info *GitRepositoryInfo) (*GitRepositoryInfo, error) {
        trimPath := strings.TrimSuffix(path, "/")
	trimPath := strings.TrimSuffix(path, ".git")
	arr := strings.Split(trimPath, "/")
	arrayLength := len(arr)
	if arrayLength >= 2 {
		info.Organisation = arr[arrayLength-2]
		info.Project = arr[arrayLength-2]
		info.Name = arr[arrayLength-1]

		return info, nil
	}

	return info, fmt.Errorf("Invalid path %s could not determine organisation and repository name", path)
}
```

### CODE REVIEW

File: pkg/jx/cmd/common_environments.go [hunk 1]
```diff
@@ -25,7 +25,8 @@ func (o *CommonOptions) createEnvironmentPullRequest(env *v1.Environment, modify
 	if gitURL == "" {
 		return answer, fmt.Errorf("No source git URL")
 	}
-	gitInfo, err := gits.ParseGitURL(gitURL, false)
+	hasSCM := strings.Index(gitURL, "/scm/") != -1
```
Comment (jstrachan) [inline pkg/jx/cmd/common_environments.go]: btw why do we care if a gitURL has `/scm/` inside it?

Comment (sanketjpatel) [reply pkg/jx/cmd/common_environments.go]: The bitbucketserver git urls are of the form
```http://test-user@example.bitbucketserver.com/scm/test-org/node-http.git```

while others are of the form ```https://github.com/jenkins-x/jx.git```

I think I took a wrong approach to solve this problem.
There is a function parsePath https://github.com/jenkins-x/jx/blob/master/pkg/gits/git_url.go#L156

```
func parsePath(path string, hasSCM bool, info *GitRepositoryInfo) (*GitRepositoryInfo, error) {
	arr := strings.Split(strings.TrimPrefix(path, "/"), "/")
	if len(arr) >= 2 {
		if hasSCM {
			info.Organisation = arr[1]
			info.Project = arr[1]
			info.Name = strings.TrimSuffix(arr[2], ".git")
		} else {
			info.Organisation = arr[0]
			info.Name = strings.TrimSuffix(arr[1], ".git")
		}

		return info, nil
	} else {
		return info, fmt.Errorf("Invalid path %s could not determine organisation and repository name", path)
	}
}
```

Could have avoided sending the `hasSCM` parameter and updated the method as follows:

```
func parsePath(path string, info *GitRepositoryInfo) (*GitRepositoryInfo, error) {
	trimPath := strings.TrimSuffix(path, ".git")
	arr := strings.Split(trimPath, "/")
	arrayLength := len(arr)
	if arrayLength >= 2 {
		info.Organisation = arr[arrayLength-2]
		info.Project = arr[arrayLength-2]
		info.Name = arr[arrayLength-1]

		return info, nil
	}

	return info, fmt.Errorf("Invalid path %s could not determine organisation and repository name", path)
}
```

Let me update the PR so we can avoid having this mess sprawl over, sorry about introducing it.

**Comment by michaelneale** [comment]
this is very cool to see !

**Comment by jstrachan** [comment]
much cleaner - thanks!

**Pull request merged**